EIPS/eip-5548.md

```diff
@@ -0,0 +1,126 @@
+
+This is intended for NFT creators who wish to exercise some control over where their creations may be bought and sold, without limiting the rights of token holders. 
+
+There are many reasons why an NFT creator may wish to exercise control over which intermediary smart-contracts may buy and sell their creations on an owner's behalf. An intermediary platform may present artwork in a manner which the creator finds objectionable or dishonest, or the intermediary platform may openly disregard off-chain codes, conventions, or practices which the creator considers to be an essential part of the artwork or their career.
+
+At the time of writing, several intermediary platforms have chosen to disregard the widely agreed-upon convention of remitting resale royalties to creators. It is expected that the extension proposed in this EIP will frequently be paired with [EIP-2981](./eip-2981.md), with EIP-2981 acting as a way of signalling standardized royalty information, and this EIP acting as a method of denying non-compliant intermediary platforms.
+
+
+## Motivation
+
+The promise of creator empowerment has driven adoption of the NFT standard (and of the Ethereum blockchain in general) for artists and creators. The opportunity to cut out intermediaries, or "middlemen," is a central tenet not just of NFT culture, but of crypto culture in general. Nonetheless, intermediaries persist. In many cases, these intermediaries serve an important function, creating smoother experiences for end users, or presenting NFT metadata in easily consumable formats. 
+
+However, as intermediary platforms willfully circumvent social codes established by NFT creators, it is within the spirit of crypto to provide technological tools to deny allowances to those intermediaries.
+
+It should be noted that this extension will not (nor does it aim to) stamp out any nefarious behaviour (such as royalty-dodging) from token holders. The position of this EIP is that any restriction to owner's ability to `Transfer` their NFTs would be overreaching. And of course, denyListed platforms could deploy new smart contracts, placing creators in the position of playing "whack-a-mole" to add new deployments to the denyList. 
+
+Nonetheless, there is benefit to creating some punitive friction for bad actors, and to implementing an EIP-721 extension which signals creator's intention to exert control over the monetization of their creations.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+Our denyList is a single new mapping of operator addresses, paired with true/false values. An event is emitted when an address is added to the denyList.
+
+```solidity
+
+mapping (address => bool) public denyList;
+event OperatorDenied(address badOperator);
```

Comment:
you need to add the boolean if you want the `updateDenyList` to allow the ability to remove an operator from the deny list as well. otherwise it will emit the `OperatorDenied` event even when they are being "allowed".

Reply:
Right... I think it's good to allow previously denied addresses to be readmitted. Suppose a situation where the creator and the platform were willing to work out an agreement or -- wild thought here -- the offending operator actually changed its policy.

So I guess the `OperatorDenied` event should be renamed `denyListUpdated` or  something like that.

Reply:
Added in latest commit.